wallets/core/tests/providers.test.ts

```diff
@@ -0,0 +1,210 @@
+import type { Context } from '../src/hub/namespaces/mod.js';
+import type { EvmActions } from '../src/namespaces/evm/types.js';
+import type { SolanaActions } from '../src/namespaces/solana/types.js';
+import type {
+  AccountsWithActiveChain,
+  CaipAccount,
+} from '../src/types/accounts.js';
+import type { FunctionWithContext } from '../src/types/actions.js';
+
+import { describe, expect, test, vi } from 'vitest';
+
+import {
+  ActionBuilder,
+  NamespaceBuilder,
+  ProviderBuilder,
+} from '../src/builders/mod.js';
+import { garbageWalletInfo } from '../src/test-utils/fixtures.js';
+
+describe('check Provider works with Blockchain correctly', () => {
+  const walletName = 'garbage-wallet';
+
+  test('connect successfully when two blockchain type has been added to Provider', async () => {
+    // Wallet Code
+    const spyOnEvmConnect = vi.fn();
+    const evmConnect: FunctionWithContext<
+      EvmActions['connect'],
+      Context
+    > = async (_context, _chain) => {
+      spyOnEvmConnect();
+      return {
+        accounts: ['eip155:0x1:0x000000000000000000000000000000000000dead'],
+        network: 'eth',
+      };
+    };
+
+    const spyOnSolanaConnect = vi.fn();
+    const solanaConnect: FunctionWithContext<
+      SolanaActions['connect'],
+      Context
+    > = async () => {
+      spyOnSolanaConnect();
+      return ['solana:mainnet:1nc1nerator11111111111111111111111111111111'];
+    };
+
+    const evmProvider = new NamespaceBuilder<EvmActions>('eip155', walletName)
+      .action('connect', evmConnect)
+      .build();
+    const solanaProvider = new NamespaceBuilder<SolanaActions>(
+      'solana',
+      walletName
+    )
+      .action('connect', solanaConnect)
+      .build();
+
+    const garbageWalletBuilder = new ProviderBuilder(walletName).config(
+      'info',
+      garbageWalletInfo
+    );
+    garbageWalletBuilder.add('evm', evmProvider);
+    garbageWalletBuilder.add('solana', solanaProvider);
+
+    const garbageWallet = garbageWalletBuilder.build();
```

Comment:
We can rename, but it needs to update other places as well. Technically, `provider` is a general term and in our case it will be a wallet almost always. I think we can ignore for now. But if you think it makes it better, I can change them.